files/usr/share/cinnamon/cinnamon-settings/bin/TreeListWidgets.py

```diff
@@ -123,9 +123,38 @@ def __init__(self, label=None, columns=None, height=200, size_group=None, dep_ke
         types = []
         tv_columns = []
         for i in range(len(columns)):
-            types.append(VARIABLE_TYPE_MAP[columns[i]["type"]])
-            renderer = Gtk.CellRendererText()
-            column = Gtk.TreeViewColumn(columns[i]["title"], renderer, text=i)
+            column_def = columns[i]
+            types.append(VARIABLE_TYPE_MAP[column_def['type']])
+
+            has_option_map = 'options' in column_def and isinstance(column_def['options'], dict)
+            render_type = 'string' if has_option_map else column_def['type']
+
+            if render_type == 'boolean':
+                renderer = Gtk.CellRendererToggle()
```

Comment:
I connected the **toggled** event to the **Gtk.CellRendererToggle**.

```suggestion
                renderer = Gtk.CellRendererToggle()
                renderer.connect("toggled", self.checkbox_toggled_cb, i)
```

And this is the **checkbox_toggled_cb** function:

```python
# col is the i index passed to the renderer connection.
def checkbox_toggled_cb(self, cell, path, col):
    if path is not None:
        iter = self.model.get_iter(path)
        self.model[iter][col] = not self.model[iter][col]

        self.list_changed()
```